addons/account/models/account_bank_statement.py

```diff
@@ -575,156 +585,55 @@ def _seek_for_lines(self):
-                amounts[company_currency.id] = journal_currency._convert(amounts[currency_id], company_currency, journal.company_id, self.date)
-            if statement_line_rate:
-                amounts[foreign_currency.id] = amounts[currency_id] * statement_line_rate
-        elif currency_id == foreign_currency.id and self.foreign_currency_id:
-            if statement_line_rate:
-                amounts[journal_currency.id] = amounts[foreign_currency.id] / statement_line_rate
-                if foreign_currency != company_currency:
-                    amounts[company_currency.id] = journal_currency._convert(amounts[journal_currency.id], company_currency, journal.company_id, self.date)
-        else:
-            amounts[journal_currency.id] = company_currency._convert(amounts[company_currency.id], journal_currency, journal.company_id, self.date)
-            if statement_line_rate:
-                amounts[foreign_currency.id] = amounts[journal_currency.id] * statement_line_rate
+        rate_journal_trans_curr = journal_amount and abs(transaction_amount) / abs(journal_amount)
+        rate_comp_journal_curr = company_amount and abs(journal_amount) / abs(company_amount)
 
-        if foreign_currency == company_currency and journal_currency != company_currency and self.foreign_currency_id:
-            balance = amounts[foreign_currency.id]
-        else:
-            balance = amounts[company_currency.id]
-
-        if foreign_currency != company_currency and self.foreign_currency_id:
-            amount_currency = amounts[foreign_currency.id]
-            currency_id = foreign_currency.id
-        elif journal_currency != company_currency and not self.foreign_currency_id:
-            amount_currency = amounts[journal_currency.id]
-            currency_id = journal_currency.id
+        if currency == trans_currency:
+            trans_amount_currency = amount_currency
+            if rate_journal_trans_curr:
+                journ_amount_currency = journal_currency.round(trans_amount_currency / rate_journal_trans_curr)
+            else:
+                journ_amount_currency = 0.0
+            if rate_comp_journal_curr:
+                new_balance = company_currency.round(journ_amount_currency / rate_comp_journal_curr)
+            else:
+                new_balance = 0.0
+        elif currency == journal_currency:
+            trans_amount_currency = trans_currency.round(amount_currency * rate_journal_trans_curr)
+            if rate_comp_journal_curr:
+                new_balance = company_currency.round(amount_currency / rate_comp_journal_curr)
+            else:
+                new_balance = 0.0
         else:
-            amount_currency = amounts[company_currency.id]
-            currency_id = company_currency.id
+            journ_amount_currency = journal_currency.round(balance * rate_comp_journal_curr)
+            trans_amount_currency = trans_currency.round(journ_amount_currency * rate_journal_trans_curr)
```

Comment:
All amounts are suggested to the user and are still editable to fix such error. I think it's more logical like this because in journal entries, all conversion are already made based on rounded values